### DISCUSSION TITLE
Full class name stored for process_type in db_dbnode instead of entry point. Manual fix possible?

### DISCUSSION THREAD

**Discussion by dev-zero** [aiidateam/aiida-core#5208] [Q&A] [answered]
It seems that I made a mistake at some point wrt an entry point specification in my `setup.json` in the sense that I did not specify the full path but the imported path (details for that below). Now I have in my `db_dbnode` as the `process_type` the full class name `aiida_cp2k.workchains.my.MyWorkchain` rather than the entry point name `aiida.workflows:cp2k.my` AiiDA uses when the entry point is correctly specified:

```
aiida=# select process_type from db_dbnode where id = 58676;
                process_type
--------------------------------------------
 aiida_cp2k.workchains.my.MyWorkchain
```

When I fix the entry point the queries the QueryBuilder generates when doing:

```python
qb = QueryBuilder()
qb.append(WorkflowFactory("cp2k.my", project=["*"])
qb.all()
```

now include the entry point name rather than the full class name:

```sql
SELECT db_dbnode_1.id AS db_dbnode_1_id, db_dbnode_1.uuid AS db_dbnode_1_uuid, db_dbnode_1.node_type AS db_dbnode_1_node_type, db_dbnode_1.process_type AS db_dbnode_1_process_type, db_dbnode_1.label AS db_dbnode_1_label, db_dbnode_1.description AS db_dbnode_1_description, db_dbnode_1.ctime AS db_dbnode_1_ctime, db_dbnode_1.mtime AS db_dbnode_1_mtime, db_dbnode_1.attributes AS db_dbnode_1_attributes, db_dbnode_1.extras AS db_dbnode_1_extras, db_dbnode_1.repository_metadata AS db_dbnode_1_repository_metadata, db_dbnode_1.user_id AS db_dbnode_1_user_id, db_dbnode_1.dbcomputer_id AS db_dbnode_1_dbcomputer_id
    FROM db_dbnode AS db_dbnode_1
    WHERE CAST(db_dbnode_1.node_type AS VARCHAR) LIKE 'process.workflow.workchain.%'
    AND
    (db_dbnode_1.process_type = 'aiida.workflows:cp2k.my' OR CAST(db_dbnode_1.process_type AS VARCHAR) LIKE 'aiida.workflows:cp2k.my.')
```

Which of course breaks my queries. Can I simply do an SQL update statement to fix the `process_type`?

### Details about the mistake
The mistake in my entry point specification was that I have the following project setup:

```
aiida_cp2k/workchains/__init__.py
aiida_cp2k/workchains/my.py
```

and in `aiida_cp2k/workchains/__init__.py` I have:

```python
from .my import MyWorkchain
```

then I used as my entry point spec:

```json
    "entry_points": {
        "aiida.workflows": [
            "cp2k.my = aiida_cp2k.workchains:MyWorkChain",
        ]
    }
```

which at some version of AiiDA started to give me the warning:

```
/scratch/tiziano/work/aiida/aiida-core/aiida/orm/querybuilder.py:1328: AiidaEntryPointWarning: Process type
'aiida_cp2k.workchains.my.MyWorkchain' does not correspond to a registered entry.
This risks queries to fail once the location of the process class changes.
Add an entry point for 'aiida_cp2k.workchains.my.MyWorkchain' to remove this warning.
```

Thanks in advance!

**Comment by zhubonan** [comment, marked as answer]
I think you can just do a SQL update - seems to be the only way out, I have used it to update computer hostnames etc.. 
Otherwise, AiiDA also have a low-level interface which may allow you to update some of the attributes (`Node._backend_entity`)? I don't have any experience with it though.

**Reply by dev-zero** [reply]
Thank you very much!
